python/ribasim/ribasim/validation.py

```diff
@@ -1,85 +1,124 @@
-    "Terminal": [],
-    "DiscreteControl": [
         "Pump",
+    ],
+    "LevelBoundary": [
         "Outlet",
         "TabulatedRatingCurve",
         "LinearResistance",
-        "ManningResistance",
-        "PidControl",
+        "Pump",
+    ],
+    "Pump": [
+        "LevelBoundary",
+        "Basin",
+        "Terminal",
+    ],
+    "UserDemand": [
+        "LevelBoundary",
+        "Basin",
+        "Terminal",
+    ],
+    "TabulatedRatingCurve": [
+        "LevelBoundary",
+        "Basin",
+        "Terminal",
     ],
-    "ContinuousControl": ["Pump", "Outlet"],
-    "PidControl": ["Pump", "Outlet"],
-    "UserDemand": ["Basin", "Terminal", "LevelBoundary"],
-    "LevelDemand": ["Basin"],
     "FlowDemand": [
         "LinearResistance",
+        "Outlet",
+        "TabulatedRatingCurve",
         "ManningResistance",
+        "Pump",
+    ],
+    "FlowBoundary": [
+        "LevelBoundary",
+        "Basin",
+        "Terminal",
+    ],
+    "Basin": [
+        "LinearResistance",
+        "UserDemand",
+        "Outlet",
         "TabulatedRatingCurve",
+        "ManningResistance",
         "Pump",
+    ],
+    "ManningResistance": [
+        "Basin",
+    ],
+    "LevelDemand": [
+        "Basin",
+    ],
+    "DiscreteControl": [
+        "LinearResistance",
+        "PidControl",
         "Outlet",
+        "TabulatedRatingCurve",
+        "ManningResistance",
+        "Pump",
+    ],
+    "Outlet": [
+        "LevelBoundary",
+        "Basin",
+        "Terminal",
+    ],
+    "ContinuousControl": [
+        "Outlet",
+        "Pump",
     ],
+    "LinearResistance": [
+        "LevelBoundary",
+        "Basin",
+    ],
+    "Terminal": [],
 }
 
 
-# Function to validate connectivity between two node types
-def can_connect(node_type_up: str, node_type_down: str) -> bool:
-    if node_type_up in node_type_connectivity:
-        return node_type_down in node_type_connectivity[node_type_up]
+# Function to validate connection
+def can_connect(node_up: str, node_down: str) -> bool:
+    if node_up in node_type_connectivity:
+        return node_down in node_type_connectivity[node_up]
     return False
 
 
 flow_edge_neighbor_amount: dict[str, list[int]] = {
-    # list[int] = [in_min, in_max, out_min, out_max]
-    "Basin": [0, int(1e9), 0, int(1e9)],
-    "LinearResistance": [1, 1, 1, 1],
-    "ManningResistance": [1, 1, 1, 1],
-    "TabulatedRatingCurve": [1, 1, 1, int(1e9)],
-    "LevelBoundary": [0, int(1e9), 0, int(1e9)],
-    "FlowBoundary": [0, 0, 1, int(1e9)],
-    "Pump": [1, 1, 1, int(1e9)],
-    "Outlet": [1, 1, 1, 1],
-    "Terminal": [1, int(1e9), 0, 0],
-    "DiscreteControl": [0, 0, 0, 0],
-    "ContinuousControl": [0, 0, 0, 0],
     "PidControl": [0, 0, 0, 0],
+    "LevelBoundary": [0, 9223372036854775807, 0, 9223372036854775807],
+    "Pump": [1, 1, 1, 1],
     "UserDemand": [1, 1, 1, 1],
-    "LevelDemand": [0, 0, 0, 0],
+    "TabulatedRatingCurve": [1, 1, 1, 1],
```

Comment:
This is different than before. Did we make a mistake previously?

Reply:
No I had to remove support for multiple outflows for TabulatedRatingCurve and Pump to do max downstream level.